miscellaneous/roboconf-integration-tests-dm-with-agents-in-memory/src/test/java/net/roboconf/integration/tests/dm/with/agents/in/memory/ExecuteScriptResourcesTest.java

```diff
@@ -0,0 +1,149 @@
+@ExamReactorStrategy( PerMethod.class )
+public class ExecuteScriptResourcesTest extends DmWithAgentInMemoryTest {
+
+	private static final String APP_LOCATION = "my.app.location";
+
+	@ProbeBuilder
+	public TestProbeBuilder probeConfiguration( TestProbeBuilder probe ) {
+
+		// We need to specify the classes we need
+		// and that come from external modules.
+		probe.addTest( DmWithAgentInMemoryTest.class );
+		probe.addTest( TestUtils.class );
+
+		probe.addTest( AbstractIntegrationTest.class );
+		probe.addTest( IMessagingConfiguration.class );
+		probe.addTest( ItConfigurationBean.class );
+
+		probe.addTest( MyHandler.class );
+		probe.addTest( MyTargetResolver.class );
+
+		return probe;
+	}
+
+
+	@Override
+	@Configuration
+	public Option[] config() throws Exception {
+
+		File resourcesDirectory = TestUtils.findApplicationDirectory( "simple" );
+		String appLocation = resourcesDirectory.getAbsolutePath();
+		return OptionUtils.combine(
+				super.config(),
+				systemProperty( APP_LOCATION ).value( appLocation ));
+	}
+
+
+	@Test
+	public void run() throws Exception {
+
+		// Update the manager
+		configureManagerForInMemoryUsage();
+
+		// Load the application
+		String appLocation = System.getProperty( APP_LOCATION );
+		ApplicationTemplate tpl = this.manager.applicationTemplateMngr().loadApplicationTemplate( new File( appLocation ));
+		ManagedApplication ma = this.manager.applicationMngr().createApplication( "test", null, tpl );
+		Assert.assertNotNull( ma );
+		Assert.assertEquals( 1, this.manager.applicationMngr().getManagedApplications().size());
+
+		// Create script files
+		String targetId = this.manager.targetsMngr().findTargetId(ma.getApplication(), "/MySQL VM");
+		File dir = new File( this.manager.configurationMngr().getWorkingDirectory(), ConfigurationUtils.TARGETS + "/" + targetId );
+		Utils.writeStringInto( "#!/bin/bash\necho toto > toto.txt", new File( dir, "totoscript-all.sh"));
+		Utils.writeStringInto( "Bonjour le monde cruel", new File( dir, "titiscript.py"));
+
+		// Deploy
+		Instance mysql = InstanceHelpers.findInstanceByPath( ma.getApplication(), "/MySQL VM/MySQL" );
+		Assert.assertNotNull( mysql );
+
+		this.manager.instancesMngr().deployAndStartAll(ma, null);
+
+		// The deploy and start messages for 'app' and 'MySQL' were stored in the DM.
+		// Wait for them to be picked up by the message checker thread.
+		// 7s = 6s (Manager#TIMER_PERIOD) + 1s for security
+		Thread.sleep( 7000 );
+
+		this.manager.instancesMngr().changeInstanceState( ma, mysql.getParent(), InstanceStatus.DEPLOYED_STARTED );
+		Assert.assertEquals( InstanceStatus.DEPLOYED_STARTED, mysql.getParent().getStatus());
+
+		// Verify that the main script is executed
+		File vmDir = InstanceHelpers.findInstanceDirectoryOnAgent( mysql.getParent() );
+		File toto = new File( vmDir, "toto.txt" );
+		Assert.assertTrue( toto.exists() );
+
+		String s = Utils.readFileContent( toto );
+		List<File> files = Utils.listAllFiles(vmDir);
```

Comment:
- List files
- Verify the number of files
- Verify the file names

- Read a file content
- Verify its content :wink: